pkg/types/installconfig.go

```diff
@@ -351,3 +351,87 @@ type BootstrapInPlace struct {
 	// InstallationDisk is the target disk drive for coreos-installer
 	InstallationDisk string `json:"installationDisk"`
 }
+
+// ClusterCreateError defines a custom error type that would help identify where the error occurs
+// during the bootstrap phase of the installation process. This would help identify whether the error
+// comes either from the Kubernetes API failure, the bootstrap failure or a general kubernetes client
+// creation error. In the event of any error, this interface packages the error message and a custom
+// log message that must be neatly presented to the user before termination of the project.
+type ClusterCreateError interface {
+	GetError() error
+	GetLogMessage() string
+}
+
+// APIError defines a custom ClusterCreateError that indicates the API failure of the bootstrap phase.
+// Provides a relevant log message along with the actual error that occurs.
+type APIError struct {
+	errorInfo error
+}
+
+// NewAPIError creates an APIError object with the error information provided.
+func NewAPIError(errorInfo error) APIError {
+	return APIError{
+		errorInfo: errorInfo,
+	}
+}
+
+// GetError provides the actual stored error that occured during installation.
+func (api APIError) GetError() error {
+	return api.errorInfo
+}
+
+// GetLogMessage provides a custom error message relevant to the API failure that will
+// help the users understand better.
+func (api APIError) GetLogMessage() string {
+	return "failed waiting for Kubernetes API, this error usually happens when there " +
+		"is a problem on the bootstrap host that prevents creating a temporary control plane"
+}
+
+// BootstrapError defines a custom ClusterCreateError that indicates the bootstrap failure.
+// Provides a relevant log message along with the actual error that occurs.
+type BootstrapError struct {
+	errorInfo error
+}
+
+// NewBootstrapError creates a BootstrapError object with the error information provided.
+func NewBootstrapError(errorInfo error) BootstrapError {
+	return BootstrapError{
+		errorInfo: errorInfo,
+	}
+}
+
+// GetError provides the actual stored error that occured during installation.
+func (b BootstrapError) GetError() error {
+	return b.errorInfo
+}
+
+// GetLogMessage provides a custom error message relevant to the bootstrap failure that will
+// help the users understand better.
+func (b BootstrapError) GetLogMessage() string {
+	return "failed to wait for bootstrapping to complete, this error usually happens when there is a problem with control plane hosts that prevents " +
+		"the control plane operators to successfully run the control plane."
```

Comment:
Let's say "creating the control plane" to keep it consistent with the cause detail in `APIError`.
```suggestion
		"the control plane operators from creating the control plane."
```